.github/workflows/app-publish.yaml

```
@@ -0,0 +1,121 @@
+name: App Publish
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - release*
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    if: ${{ startsWith(github.ref, 'refs/tags/v') || github.repository == 'jellyfin/jellyfin-android' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Setup Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set JELLYFIN_VERSION
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: echo "JELLYFIN_VERSION=$(echo ${GITHUB_REF#refs/tags/v} | tr / -)" >> $GITHUB_ENV
+
+      - name: Build Release
+        run: ./gradlew --no-daemon --build-cache assembleRelease versionTxt
+
+      - name: Split APK release types
+        run: |
+          mkdir app/build/outputs/apk/libre_release app/build/outputs/apk/proprietary_release;
+          mv app/build/outputs/apk/**/jellyfin-android-*-libre-release.apk app/build/outputs/apk/libre_release/;
+          mv app/build/outputs/apk/**/jellyfin-android-*-proprietary-release.apk app/build/outputs/apk/proprietary_release/;
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: build-publish-artifact
+          retention-days: 30
+          if-no-files-found: error
+          path: |
+            app/build/outputs/apk/libre_release/*
+            app/build/outputs/apk/proprietary_release/*
+            app/build/version.txt
+
+  publish-release:
+    runs-on: ubuntu-20.04
+    needs: build-publish
+    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
```

```suggestion
    environment: release
    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
```

The secrets will be added to the release environment